pkg/reconciler/v1alpha1/revision/revision_test.go

```diff
@@ -709,17 +773,34 @@ func TestGlobalResyncOnConfigMapUpdate(t *testing.T) {
 				"queueSidecarImage": "myAwesomeQueueImage",
 			},
 		},
-		wasUpdated: func(expected string, revision *v1alpha1.Revision, deployment *appsv1.Deployment) (string, bool) {
-			var got string
-			for _, c := range deployment.Spec.Template.Spec.Containers {
-				if c.Name == resources.QueueContainerName {
-					got = c.Image
-					if got == expected {
-						return got, true
+		resource: "deployments",
+		callback: func(t *testing.T, expected string, h *Hooks, fake *kubetesting.Fake) {
+			check := func(obj runtime.Object) HookResult {
+				deployment := obj.(*appsv1.Deployment)
+				t.Logf("Deployment updated: %v", deployment.Name)
+
+				var got string
+				for _, c := range deployment.Spec.Template.Spec.Containers {
+					if c.Name == resources.QueueContainerName {
+						got = c.Image
+						if got == expected {
+							return HookComplete
+						}
 					}
 				}
+
+				t.Logf("No update occurred; expected: %s got: %s", expected, got)
+				return HookIncomplete
 			}
-			return got, false
+
+			h.OnCreate(fake, "deployments", func(obj runtime.Object) HookResult {
+				result := check(obj)
+				if result == HookIncomplete {
+					t.Log("Registering OnUpdate hook")
+					h.OnUpdate(fake, "deployments", check)
```

Comment:
This is strange, I wouldn't expect us to register a hook in another hook.  Could we just always register Create and Update?

Reply:
Sometimes only the Create fires, with the expected change. Other times the Create fires without the expected change, but then the Update fires with the expected change.

We could do both, but then WaitForHooks wouldn't work :/